File: auth0/src/test/java/com/auth0/android/provider/AuthenticationActivityMock.java
package com.auth0.android.provider;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

/**
 * Created by lbalmaceda on 6/12/17.
 */

public class AuthenticationActivityMock extends AuthenticationActivity {

    private CustomTabsController customTabsController;
    private Intent deliveredIntent;

    @Override
    protected CustomTabsController createCustomTabsController(@NonNull Context context, @Nullable String[] browserPackages) {
        return customTabsController;
    }

    @Override
    protected void deliverAuthenticationResult(Intent result) {
        this.deliveredIntent = result;
        super.deliverAuthenticationResult(result);
    }

    public void setCustomTabsController(CustomTabsController customTabsController) {
        this.customTabsController = customTabsController;
    }

    public Intent getDeliveredIntent() {
        return deliveredIntent;
    }
}

File: auth0/src/test/java/com/auth0/android/provider/AuthenticationActivityMock.kt
package com.auth0.android.provider

import android.content.Context
import android.content.Intent
import com.auth0.android.authentication.AuthenticationException

public class AuthenticationActivityMock : AuthenticationActivity() {
    internal var customTabsController: CustomTabsController? = null
    public var deliveredIntent: Intent? = null
        private set
    public var deliveredException: AuthenticationException? = null
        private set

    override fun createCustomTabsController(
        context: Context,
        options: CustomTabsOptions
    ): CustomTabsController {
        return customTabsController!!
    }

    override fun deliverAuthenticationResult(result: Intent?) {
        deliveredIntent = result
        super.deliverAuthenticationResult(result)
    }

    override fun deliverAuthenticationFailure(ex: AuthenticationException) {
        deliveredException = ex
        super.deliverAuthenticationFailure(ex)
    }
}